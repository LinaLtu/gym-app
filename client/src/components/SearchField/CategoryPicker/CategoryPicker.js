import React, { useState } from 'react';
import './CategoryPicker.scss';

const CategoryPick = ({ setItemsByCategory }) => {
    const defaultCategories = ['C1', 'C2', 'C3', 'C4', 'C5', 'C6', 'C7'];

    const [selectedCategories] = useState([])

    const handleSelect = (category) => {
        // if an item is selected
        if (!selectedCategories.includes(category)) {
            selectedCategories.push(category)
            // in an item us unselected
        } else {
            const index = selectedCategories.indexOf(category);
            selectedCategories.splice(index, 1);
        }

        setItemsByCategory(selectedCategories);
    }

    const CSS_NAME = 'category-picker';

    return (
        <div className={CSS_NAME}>
            {defaultCategories.map((category, index) => {
                return (<div onClick={() => handleSelect(category)} key={index}>
                    <input type="checkbox" id="category" name={category} />
                    <label for={category}>{category}</label>
                </div>)
            })}
        </div >
    )
}

export default CategoryPick; 