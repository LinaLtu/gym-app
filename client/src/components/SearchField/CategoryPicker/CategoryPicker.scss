.category-picker {
    display: flex;
    max-height: 50px;
}

.category-picker > div {
    padding: 5px;
}