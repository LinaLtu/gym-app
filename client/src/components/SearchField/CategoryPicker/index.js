import CategoryPick from './CategoryPicker';

export default CategoryPick;