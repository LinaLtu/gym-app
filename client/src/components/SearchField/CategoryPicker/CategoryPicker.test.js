import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import CategoryPicker from './';

Enzyme.configure({ adapter: new Adapter() })

const setCategories = jest.fn();

describe('CategoryPicker Component', () => {
    describe('Rendering', () => {
        it('renders according to design', () => {
            const component = renderer.create(
                <CategoryPicker setCategories={setCategories} />
            );
            const tree = component.toJSON();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Integration', () => {
        describe('when a category is selected', () => {
            let wrapper
            beforeEach(() => {
                wrapper = shallow(<CategoryPicker callback={setCategories} />);

            });

            it('#setCategories callback gets called', () => {
                wrapper.find('input').first().simulate('click', { target: { name: 'c1' } });
                expect(setCategories).toHaveBeenCalled();
            });
        });
    });
});
