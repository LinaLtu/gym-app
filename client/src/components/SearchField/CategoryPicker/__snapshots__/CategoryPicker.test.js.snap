// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPicker Component Rendering renders according to design 1`] = `
<div
  className="category-picker"
>
  <div>
    <input
      checked={true}
      id="category"
      name="c1"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c1"
    >
      c1
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c2"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c2"
    >
      c2
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c3"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c3"
    >
      c3
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c4"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c4"
    >
      c4
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c5"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c5"
    >
      c5
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c6"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c6"
    >
      c6
    </label>
  </div>
  <div>
    <input
      checked={true}
      id="category"
      name="c7"
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <label
      for="c7"
    >
      c7
    </label>
  </div>
</div>
`;
